src/web/app/pages-admin/admin-notifications-page/admin-notifications-page.component.spec.ts

```diff
@@ -1,30 +1,248 @@
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should display warning when attempts to edit another notification when form is open', async () => {
+    const promise: Promise<void> = Promise.resolve();
+    const modalSpy: SpyInstance = jest.spyOn(simpleModalService, 'openConfirmationModal')
+        .mockReturnValue(createMockNgbModalRef({}, promise));
+    jest.spyOn(notificationService, 'getNotifications').mockReturnValue(of({
+      notifications: [EXAMPLE_NOTIFICATION_ONE, EXAMPLE_NOTIFICATION_TWO],
+    }));
+    component.initNotificationEditFormModel();
+    component.loadNotifications();
+    component.loadNotificationEditForm(component.notificationsTableRowModels[0].notification);
+    fixture.detectChanges();
+    expect(component.isNotificationEditFormExpanded).toBeTruthy();
+
+    const expectedMessage: string = component.notificationsTableRowModels[0].notification.message;
+
+    component.loadNotificationEditFormHandler(component.notificationsTableRowModels[0].notification);
+    component.notificationEditFormModel.message = 'new message';
+    expect(component.notificationEditFormModel.message).toEqual('new message');
+
+    // attempts to load edit form for another notification
+    component.loadNotificationEditFormHandler(component.notificationsTableRowModels[1].notification);
+    await promise;
+    expect(component.notificationsTableRowModels[0].notification.message).toEqual(expectedMessage);
+    expect(component.notificationEditFormModel.message)
+        .toEqual(component.notificationsTableRowModels[1].notification.message);
+
+    expect(modalSpy).toHaveBeenCalledTimes(1);
+    expect(modalSpy).toHaveBeenLastCalledWith('Discard unsaved edit?',
+        SimpleModalType.WARNING,
+        'Warning: If you choose to edit another notification, any unsaved changes will be lost.');
+  });
+
+  it('should delete notfication', () => {
+    component.notificationsTableRowModels = [notificationTableRowModel1];
+    expect(component.notificationsTableRowModels.length).toEqual(1);
+    jest.spyOn(notificationService, 'deleteNotification').mockReturnValue(of({
+      message: 'Successfully deleted',
+    }));
+    const spy: SpyInstance = jest.spyOn(statusMessageService, 'showSuccessToast')
+    .mockImplementation((args: string) => {
+      expect(args).toEqual('Successfully deleted');
+    });
+
+    component.deleteNotificationHandler(EXAMPLE_NOTIFICATION_EDIT_MODEL.notificationId);
+    expect(spy).toHaveBeenCalledTimes(1);
+    expect(component.notificationsTableRowModels.length).toEqual(0);
+  });
+
+  it('should sort notification by start time', () => {
+    component.notificationsTableRowModels = [notificationTableRowModel1, notificationTableRowModel2];
+    component.notificationsTableRowModelsSortBy = SortBy.NOTIFICATION_CREATE_TIME;
+    component.sortNotificationsTableRowModelsHandler(SortBy.NOTIFICATION_START_TIME);
+    expect(component.notificationsTableRowModelsSortBy).toEqual(SortBy.NOTIFICATION_START_TIME);
```

Comment:
Is this test the exact same test as the one below in the notifications table component

Reply:
This is to test the function `sortNotificationsTableRowModelsHandler` in admin notification page specifically, would you like me to remove this test?

Reply:
Since the bottom one is a snapshot test, it does differ slightly so don't need to remove the test. I think you could add in a check for the row models (make sure that the row models are actually in sorted order) in addition to the `notificationsTableRowModelsSortBy` check.

Reply:
Thanks for the suggestion! I have added a check below for it.